client/src/components/Login/index.tsx

```diff
@@ -1,14 +1,32 @@
-import React from 'react'
+import React, {useEffect} from 'react'
 import { SocialLoginBtn } from '../SocialLoginBtn/index'
 import './login.scss'
+import axios from 'axios'
+import queryParser from 'query-parser-url'
+import {socialSites} from './config'
+
+const socialLoginBtnList = socialSites.map((site) => {
+  return <SocialLoginBtn icon={site.icon} text={site.text} href={site.href}/>
+})
 
 export const Login = () => {
+  useEffect(() => {
+    const url = window.location.search.substring(1)
+
+    const queryObject = queryParser(url)
+    const code = queryObject['code']
+    if (!code) return
+
+    axios.get(`/api/github-login/?code=${code}`).then(res => {
+      localStorage.setItem('token', res.data.token)
+    })
+  }, []);
+
   return (
     <div className="login-section">
       <img className="logo" src="./images/bmartlogo.jpeg" />
       <div className="social-login-wrap">
-        <SocialLoginBtn icon="github" text="Login with Github" />
-        <SocialLoginBtn icon="facebook" text="Login with Facebook" />
+        {socialLoginBtnList}
```

Comment:
jsx 부분이 분리되어 있어서 조금 어색한 느낌이 있는거 같은데
안쪽으로 넣는 건 어떻게 생각하시나요?
```suggestion
        {socialSites.map((site) => <SocialLoginBtn icon={site.icon} text={site.text} href={site.href}/>)}
```

Reply:
props를 내려줄때 구조분해를 활동하시면 더 간결해질 수 있습니다!
```suggestion
        {socialSites.map((site) => <SocialLoginBtn {...site}/>)}
```

Reply:
> jsx 부분이 분리되어 있어서 조금 어색한 느낌이 있는거 같은데
> 안쪽으로 넣는 건 어떻게 생각하시나요?
> ```suggestion
>         {socialSites.map((site) => <SocialLoginBtn icon={site.icon} text={site.text} href={site.href}/>)}
> ```

흠 view가 순수하면 좋겠다고 생각해서 분리했는데 react에서는 안에 넣나요?

Reply:
https://ko.reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator

리액트 공식문서에는 두 방법 모두 있고, 선택의 차이라고 하네요!
저는 개인적으로 인라인으로 들어가있는게 직관적이라고 생각하는데
지환님 의견은 어떠신가요?

Reply:
> props를 내려줄때 구조분해를 활동하시면 더 간결해질 수 있습니다!

저렇게 할 수도 있군요!
영지님은 어떤 방식이 나으신가요?

Reply:
> https://ko.reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator
> 
> 리액트 공식문서에는 두 방법 모두 있고, 선택의 차이라고 하네요!
> 저는 개인적으로 인라인으로 들어가있는게 직관적이라고 생각하는데
> 지환님 의견은 어떠신가요?

저는 개인적으로 따로 분리하는게 낫다고 생각하는데 안에 넣어볼게요!